src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -312,6 +312,16 @@ public function configureOptions(OptionsResolver $resolver)
             return $choiceTranslationDomain;
         };
 
+        $choiceLabelNormalizer = function (Options $options, $choiceLabel) {
+            if (false === $choiceLabel) {
+                return function () {
```

Comment:
I afraid we can't do this in 3.4, because you'll break things for people rely on the `false` value, e.g. `false === $options['choice_label']`.

Reply:
I don't think that counts otherwise we would not be able to add or change any form option. But I'm fine to merge this PR to 4.4 only.

Reply:
we have (since 4.2) the `->setDeprecated()` method to deprecating any form option before remove it or change its value.

Tecnically this would look like a BC break to me.

Reply:
What do you propose?

Reply:
I guess deprecating to pass `false` to the `createView` method and make the normalization to `callable` before calling it, I only found one occurence: `ChoiceType::createChoiceListView()`. Also that should be targeted to 4.4.

Reply:
>  make the normalization to callable before calling it

Then there is no point in adding a deprecation as false will never be passed.

Reply:
The point is, it's a public method, hence we can't add the `callable` type-hint to the `$label` argument without adding first a deprecation warning about not passing `false`.

Make the normalization to `callable` before calling to `createView` is the intended operation to get rid of the deprecation warning.

Reply:
We talking about this change https://github.com/symfony/symfony/pull/32237/files#diff-ca1123eff9e0aa207aaec667aa16c8e8R48 right?

Reply:
It will be nice to have `?callable $label` for consistency with other arguments. 
I agree with @yceruto — we should deprecate `false`. I can help with the deprecation if you wish.

Reply:
Feel free to take over